lib/tesla/middleware/telemetry.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Tesla.Middleware.Telemetry do
+  @behaviour Tesla.Middleware
+
+  @moduledoc """
+  Send the request duration and meta-information through telemetry.
+
+  ### Example usage
+  ```
+  defmodule TelemetryHandler do
+    use GenServer
+
+    def start_link(_, _) do
+      :telemetry_app.start(nil, nil)
+      :telemetry.attach(
+        "tesla-telemetry",
+        [:tesla, :telemetry, :traffic],
+        fn ([:tesla, :telemetry, :traffic], time, meta, _config) -> # Do sth end,
+        nil)
+    end
+  end
+
+  defmodule MyClient do
```

Comment:
I'd put this definition on top and then only this lines:

```elixir
:telemetry.attach("my-tesla-telemetry", [:tesla, :telemetry, :traffic], fn event, time, meta, config ->
  # do something with the event
end)
```

And reference [telemetry](https://github.com/beam-telemetry/telemetry) as the source of documentation for further usage